vllm/utils.py

```diff
@@ -1088,3 +1114,69 @@ async def _run_task_with_lock(task: Callable, lock: asyncio.Lock, *args,
     """Utility function to run async task in a lock"""
     async with lock:
         return await task(*args, **kwargs)
+
+
+def _create_dummy_modules():
```

Comment:
This is very brittle. Anytime someone adds a new module in a any file we'd need to remember to wrap it here. Couldn't we do it somehow differently?

Reply:
I've done research and asked a few people and I have not found a different way of doing it unfortunately. I'm open for suggestions but for now I have not found a more "elegant" way of doing that.

Reply:
What about using MagicMock?
https://stackoverflow.com/a/37126323
https://docs.python.org/3/library/unittest.mock.html

As far as I understand it should automatically mock everything in the hierarchy below. We could do it only for 'habana_frameworks'.

Reply:
Unfortunately MagicMock doesn't solve the submodules issue but it highly improves visibility and mokes further dummy modules additions much simpler -> Changed origin dummy modules handling to MagicMock.

Reply:
Hmm... Perhaps something like this could work:
```python
builtin_import = __builtins__.__import__

def import_wrapper(name, *args, **kwargs):
        if 'habana_frameworks' in name:
                sys.modules[name] = MagicMock()
        return builtin_import(name, *args, **kwargs)

__builtins__.__import__ = import_wrapper
```

Could you please check if it works? (last thing, I promise! 😄 )